arkouda/dataframe.py

```diff
@@ -287,14 +294,11 @@ def __getitem__(self, key):
         # Select rows using a slice
         elif isinstance(key, slice):
             # result = DataFrame()
-            data = {}
+            rtn_data = {}
             s = key
             for k in self._columns:
-                data[k] = UserDict.__getitem__(self, k)[s.start:s.stop:s.step]
-            # result._columns.append(k)
-            # result._empty = False
-            return DataFrame(initialdata=data)
-
+                rtn_data[k] = UserDict.__getitem__(self, k)[s.start:s.stop:s.step]
+            return DataFrame(initialdata=rtn_data, index=list(range(self.size)[s.start:s.stop:s.step]))
```

Comment:
i know this is how it was done in the existing code but I don't think you need to rebuild the slice. I think something like this works
```python
UserDict.__getitem__(self, k)[s]
return DataFrame(initialdata=rtn_data, index=arange(self.size)[s])
```
I also think we should use the `ak.arange` function since a list of size `self.size` could be large and `._set_index` will just convert to a pdarray anyway

Reply:
I updated this to the requested configuration. Sorry I totally missed this when I was making changes this morning!